File: src/selection_sort.rs
/// Returns the index of the minimum element in the given slice.
fn get_min_index(last_slice: &[usize]) -> usize {
    let mut min = last_slice[0];
    let mut min_index = 0;

    // TODO: maybe skip the first?
    for (index, element) in last_slice.iter().enumerate() {
        if *element < min {
            min = *element;
            min_index = index;
        }
    }

    min_index
}

/// Sorts the given array, in place, by using the Selection sort algorithm.
pub fn sort_by_selection_sort(array: &mut [usize]) {
    for index in 0..array.len() - 1 {
        let mut min_index = get_min_index(&array[index..]);

        min_index += index;

        array.swap(index, min_index);
    }
}

File: python_src/selection_sort.py
def get_min_index(array, start):
    min_element = array[start]
    min_index = start

    for i in range(start + 1, len(array)):
        if array[i] < min_element:
            min_element = array[i]
            min_index = i

    return min_index

def sort_by_selection_sort(array):
    for i in range(len(array) - 1):
        min_index = get_min_index(array, i)

        array[i], array[min_index] = array[min_index], array[i]